source/gateway.go

```diff
@@ -224,21 +223,29 @@ func (sc *gatewaySource) setResourceLabel(config istiomodel.Config, endpoints []
 	}
 }
 
-func (sc *gatewaySource) targetsFromIstioIngressStatus() (targets endpoint.Targets, err error) {
-	if svc, e := sc.kubeClient.CoreV1().Services(sc.istioNamespace).Get(sc.istioIngressGatewayName, metav1.GetOptions{}); e != nil {
-		err = e
-	} else {
-		for _, lb := range svc.Status.LoadBalancer.Ingress {
-			if lb.IP != "" {
-				targets = append(targets, lb.IP)
-			}
-			if lb.Hostname != "" {
-				targets = append(targets, lb.Hostname)
+func (sc *gatewaySource) targetsFromIstioIngressGatewayServices() (targets endpoint.Targets, err error) {
+	for _, lbService := range sc.istioIngressGatewayServices {
+		lbNamespace, lbName, e := parseIngressGateway(lbService)
```

Comment:
can you change it to err, so that you don't have to assign e.

Reply:
Sure, I prefer that even. Didn't want to change current style. Will do!